README.md

```diff
@@ -0,0 +1,486 @@
+# sanctuary-def
+
+sanctuary-def facilitates the definition of curried JavaScript functions which
+are explicit about the number of arguments to which they may be applied and the
+types of those arguments.
+
+```javascript
+const $ = require('sanctuary-def');
+
+const def = $.create($.env);  // Explained in "Environments" section
+
+//    add :: Number -> Number -> Number
+const add =
+def('add', {}, [$.Number, $.Number, $.Number], (x, y) => x + y);
+```
+
+Applying `add` to two arguments gives the expected result:
+
+```javascript
+add(2, 2);
+// => 4
+```
+
+Applying `add` to greater than two arguments results in an exception being
+thrown:
+
+```javascript
+add(2, 2, 2);
+// ! TypeError: ‘add’ requires two arguments; received three arguments
+```
+
+Applying `add` to fewer than two arguments results in a function awaiting the
+remaining arguments. This is known as partial application. Partial application
+is convenient as it allows more specific functions to be defined in terms of
+more general ones:
+
+```javascript
+//    inc :: Number -> Number
+const inc = add(1);
+
+inc(7);
+// => 8
+```
+
```

Comment:
I'm not sure about this suggestion.  Is it worth mentioning that an error like `inc(7, 6)` will report an incorrect use of `add` instead of `inc`?

Also, would it be considered an acceptable practice to define `inc` as `const inc = def('inc', {}, [$.Number, $.Number], add(1));` if we really wanted specific function names in the error messages?  If not, it's certainly worth mentioning this as an anti-pattern.

This example is simple enough that the error should be obvious to figure out, but I'm imagining a situation where a newb dev, like myself, is reusing `inc` in a file far from its definition and getting an error unexpectedly about `add`.